src/Configuration/ValueObject/ExportConfig.php

```diff
@@ -0,0 +1,171 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractorConfig\Configuration\ValueObject;
+
+use Keboola\DbExtractorConfig\Exception\PropertyNotSetException;
+
+class ExportConfig implements ValueObject
+{
+    /** Custom export query */
+    private ?string $query;
+
+    /** Table that will be exported (if query is not set) */
+    private ?InputTable $table;
+
+    /** Configuration of incremental fetching */
+    private ?IncrementalFetchingConfig $incrementalFetchingConfig;
+
+    /** Columns that will be exported, cannot be used with query, if empty => all columns */
+    private array $columns;
+
+    /** Name of the output table */
+    private string $outputTable;
+
+    /** PK key can be composed of multiple columns, therefore an array */
+    private array $primaryKey;
+
+    /** Number of max retries if an error occurs */
+    private int $maxRetries;
+
+    public static function fromArray(array $data): self
+    {
+        return new self(
+            $data['query'],
+            empty($data['query']) ? InputTable::fromArray($data) : null,
+            empty($data['query']) ? IncrementalFetchingConfig::fromArray($data) : null,
+            $data['columns'],
+            $data['outputTable'],
+            $data['primaryKey'],
+            $data['retries']
+        );
+    }
+
+    public function __construct(
+        ?string $query,
+        ?InputTable $table,
+        ?IncrementalFetchingConfig $incrementalFetchingConfig,
+        array $columns,
+        string $outputTable,
+        array $primaryKey,
+        int $maxRetries
+    ) {
+        $query = $query !== null ? trim($query) : null;
+        $outputTable = trim($outputTable);
+        $this->query = $query;
+        $this->table = $table;
+        $this->incrementalFetchingConfig = $incrementalFetchingConfig;
+        $this->columns = $columns;
+        $this->outputTable = $outputTable;
+        $this->primaryKey = $primaryKey;
+        $this->maxRetries = $maxRetries;
+    }
+
+    public function hasTable(): bool
+    {
+        return $this->table !== null;
+    }
+
+    public function getTable(): InputTable
+    {
+        if ($this->table === null) {
+            throw new PropertyNotSetException('Table is not set, use "query".');
+        }
+
+        return $this->table;
+    }
+
+    public function isIncrementalFetching(): bool
+    {
+        return $this->incrementalFetchingConfig !== null;
+    }
+
+    public function hasIncrementalLimit(): bool
```

Comment:
spíš `hasIncrementalFetchingLimit`

Reply:
vdaka. to som prehliadol.

Reply:
fixnute v https://github.com/keboola/db-extractor-config/pull/8/commits/9299716d3a9aaaafab94e261df0cb9ee785e1c50